### PR TITLE
PARQUET-979: Limit size of min, max or disable stats for long binary types

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

File: src/parquet/column_writer.cc [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
 namespace parquet {
 
+static constexpr int MAX_STATS_SIZE = 4096;  // limit stats to 4k
```
Comment (xhochy) [inline src/parquet/column_writer.cc]: Is this defined by the Parquet standard or arbitrarily chosen?

Comment (majetideepak) [reply src/parquet/column_writer.cc]: I followed parquet-mr.
https://github.com/apache/parquet-mr/blob/0d55abd05b0e5027c18e60d1ac3b22998dd00951/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java#L88
I don't see this specified in the spec. We should probably add it there. I will open a JIRA to do this.

Comment (xhochy) [reply src/parquet/column_writer.cc]: The alternative would be to use this as sensible default and add it as an option to `WriterProperties`.

Comment (majetideepak) [reply src/parquet/column_writer.cc]: I added it to `ColumnProperties` since it gives more flexibility.

File: src/parquet/properties.h [hunk 2]
```diff
@@ -348,19 +357,23 @@ class PARQUET_EXPORT WriterProperties {
...
-    return column_properties(path).dictionary_enabled;
+    return column_properties(path).dictionary_enabled_;
   }
 
   bool statistics_enabled(const std::shared_ptr<schema::ColumnPath>& path) const {
-    return column_properties(path).statistics_enabled;
+    return column_properties(path).statistics_enabled_;
+  }
+
+  size_t max_stats_size(const std::shared_ptr<schema::ColumnPath>& path) const {
```
Comment (xhochy) [inline src/parquet/properties.h]: Please use `max_statistics_size`

**Pull request closed without merging**